### PR TITLE
feat(propagation): add DD_TRACE_PROPAGATION_STYLE envar

### PR DESCRIPTION
**Pull request by ZStriker19**
## Description
Adding a new envar: `DD_TRACE_PROPAGATION_STYLE` which essentially sets both inject and extract. This envar will also help align tracers in that there is variation in tracer envar keys for extract and inject currently.

## Motivation
This allows customers to set a single envar to configure both inject and extract in the event that they don't need to have different values.

## Design 
`DD_TRACE_PROPAGATION_STYLE` is overridden by `DD_TRACE_PROPAGATION_STYLE_EXTRACT` for propagation extraction and by `DD_TRACE_PROPAGATION_STYLE_INJECTION` for propagation injection.

## Testing strategy
Change tests to use `DD_TRACE_PROPAGATION_STYLE` and wrote new tests to test overiding behavior and interaction with `DD_TRACE_PROPAGATION_STYLE_EXTRACT` and `DD_TRACE_PROPAGATION_STYLE_INJECT`.

## Checklist
- [x] Add additional sections for `feat` and `fix` pull requests.
- [x] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: ddtrace/settings/config.py [hunk 1]
```diff
@@ -201,12 +206,19 @@ def __init__(self):
...
         # Propagation styles
-        self._propagation_style_extract = _parse_propagation_styles(
-            "DD_TRACE_PROPAGATION_STYLE_EXTRACT", default=PROPAGATION_STYLE_DATADOG
-        )
-        self._propagation_style_inject = _parse_propagation_styles(
-            "DD_TRACE_PROPAGATION_STYLE_INJECT", default=PROPAGATION_STYLE_DATADOG
+        self._propagation_style_extract = self._propagation_style_inject = _parse_propagation_styles(
+            "DD_TRACE_PROPAGATION_STYLE", default=PROPAGATION_STYLE_DATADOG
         )
+        _parse_propagation_styles("DD_TRACE_PROPAGATION_STYLE_EXTRACT", default=None)
```
Comment (majorgreys) [inline ddtrace/settings/config.py:212]: Not needed? 
```suggestion
```

Comment (ZStriker19) [reply ddtrace/settings/config.py:212]: Def not needed.

**Review by majorgreys — APPROVED** [review]

File: releasenotes/notes/add_DD_TRACE_PROPAGATION_STYLE-ee71fda484a0e444.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Added environment variable ``DD_TRACE_PROPAGATION_STYLE`` to configure both injection and extraction propagation styles. The configured styles can be overridden with environment variables ``DD_TRACE_PROPAGATION_STYLE_INJECT`` and ``DD_TRACE_PROPAGATION_STYLE_EXTRACT``.
```
Comment (mabdinur) [inline releasenotes/notes/add_DD_TRACE_PROPAGATION_STYLE-ee71fda484a0e444.yaml:4]: nit: we need a new line here but LGTM

**Review by mabdinur — APPROVED** [review]

File: releasenotes/notes/add_DD_TRACE_PROPAGATION_STYLE-ee71fda484a0e444.yaml (hunk 2 above)
Comment (ZStriker19) [inline releasenotes/notes/add_DD_TRACE_PROPAGATION_STYLE-ee71fda484a0e444.yaml:4]:
```suggestion
    Added environment variable ``DD_TRACE_PROPAGATION_STYLE`` to configure both injection and extraction propagation styles. The configured styles can be overridden with environment variables ``DD_TRACE_PROPAGATION_STYLE_INJECT`` and ``DD_TRACE_PROPAGATION_STYLE_EXTRACT``.

```

File: docs/configuration.rst [hunk 3]
```diff
@@ -209,25 +209,49 @@ below:
...
 
+   DD_TRACE_PROPAGATION_STYLE:
+     default: |
+         ``datadog``
+     description: |
+         Comma separated list of propagation styles used for extracting trace context from inbound request headers and injecting trace context into outbound request headers.
+
+         Overridden by ``DD_TRACE_PROPAGATION_STYLE_EXTRACT`` for extraction if both are set.
+
+         Overridden by ``DD_TRACE_PROPAGATION_STYLE_INJECT`` for injection if both are set.
```
[the remark this answers is not in the document]
Comment (ZStriker19) [reply docs/configuration.rst:220]: If both `DD_TRACE_PROPAGATION_STYLE` and `DD_TRACE_PROPOAGATION_STYLE_EXTRACT` are set, `DD_TRACE_PROPOAGATION_STYLE_EXTRACT` overrides for the extraction propagation style.

If both `DD_TRACE_PROPAGATION_STYLE` and `DD_TRACE_PROPOAGATION_STYLE_INJECT` are set, `DD_TRACE_PROPOAGATION_STYLE_INJECT` overrides for the injection propagation style.

File: docs/configuration.rst [hunk 4]
```diff
@@ -209,25 +209,49 @@ below:
...
+         Example: ``DD_TRACE_PROPAGATION_STYLE_EXTRACT="datadog,b3"`` to check for both ``x-datadog-*`` and ``x-b3-*``
+         headers when parsing incoming request headers for a trace context.
+
    DD_TRACE_PROPAGATION_STYLE_EXTRACT:
      default: |
          ``datadog``
      description: |
          Comma separated list of propagation styles used for extracting trace context from inbound request headers.
 
+         Overrides ``DD_TRACE_PROPAGATION_STYLE`` if both are set.
```
[the remark this answers is not in the document]
Comment (ZStriker19) [reply docs/configuration.rst:237]: I'll change to specify it specifically overrides extract.

File: docs/configuration.rst [hunk 5]
```diff
@@ -209,25 +209,49 @@ below:
...
+         headers when parsing incoming request headers for a trace context. In addition, to inject both ``x-datadog-*`` and ``x-b3-*``
+         headers into outbound requests.
 
    DD_TRACE_PROPAGATION_STYLE_INJECT:
      default: |
          ``datadog``
      description: |
          Comma separated list of propagation styles used for injecting trace context into outbound request headers.
 
+         Overrides ``DD_TRACE_PROPAGATION_STYLE`` if both are set.
```
[the remark this answers is not in the document]
Comment (ZStriker19) [reply docs/configuration.rst:253]: I'll change to specify it specifically overrides inject.

File: docs/configuration.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (ZStriker19) [reply docs/configuration.rst:220]: If you set `DD_TRACE_PROPOAGATION_STYLE_EXTRACT` it'll override for extraction. 

If you set `DD_TRACE_PROPOAGATION_STYLE_INJECT` it'll override for injection.

[the remark this answers is not in the document]
Comment (Yun-Kim) [reply docs/configuration.rst:220]:
```suggestion
         Overridden by ``DD_TRACE_PROPAGATION_STYLE_EXTRACT``.

         Overridden by ``DD_TRACE_PROPAGATION_STYLE_INJECT``.
```

File: docs/configuration.rst [hunk 6]
```diff
@@ -209,25 +209,49 @@ below:
...
+         Example: ``DD_TRACE_PROPAGATION_STYLE_EXTRACT="datadog,b3"`` to check for both ``x-datadog-*`` and ``x-b3-*``
+         headers when parsing incoming request headers for a trace context.
+
    DD_TRACE_PROPAGATION_STYLE_EXTRACT:
      default: |
          ``datadog``
      description: |
          Comma separated list of propagation styles used for extracting trace context from inbound request headers.
 
+         Overrides ``DD_TRACE_PROPAGATION_STYLE`` for extraction propagtion style if both are set.
```
Comment (Yun-Kim) [inline docs/configuration.rst:237]:
```suggestion
         Overrides ``DD_TRACE_PROPAGATION_STYLE`` for extraction propagation style.
```

File: docs/configuration.rst [hunk 7]
```diff
@@ -209,25 +209,49 @@ below:
...
+         headers when parsing incoming request headers for a trace context. In addition, to inject both ``x-datadog-*`` and ``x-b3-*``
+         headers into outbound requests.
 
    DD_TRACE_PROPAGATION_STYLE_INJECT:
      default: |
          ``datadog``
      description: |
          Comma separated list of propagation styles used for injecting trace context into outbound request headers.
 
+         Overrides ``DD_TRACE_PROPAGATION_STYLE`` for injection propagation style if both are set.
```
Comment (Yun-Kim) [inline docs/configuration.rst:253]:
```suggestion
         Overrides ``DD_TRACE_PROPAGATION_STYLE`` for injection propagation style.
```

**Review by Yun-Kim — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Review by jd — APPROVED** [review]

**Pull request merged**